src/mlpack/methods/ann/layer/highway_impl.hpp

```diff
@@ -229,7 +229,10 @@ void Highway<InputDataType, OutputDataType, CustomLayers...>::serialize(
   }
 
   ar & CEREAL_NVP(model);
-  ar & CEREAL_NVP(network);
+  for (size_t i = 0; i < network.size(); ++i) 
+  {
+   ar & CEREAL_VARIANT_POINTER(network.at(i));
+  }
```

Comment:
I imagine, in this case, that we have to also serialize `network.size()` and then call `resize()` to set the correct size if we are loading. :+1:

Reply:
@rcurtin Of course,  I did not finish this yet, I was just able to serialize all layers and network types.
This should follow the same principle as `CEREAL_VECTOR`.
I just wanted to be sure that the variant wrapper is working in the first place.

Reply:
Considering the `boost::string_view`, I was thinking of creating a wrapper that converts `boost::string_view` into `std::string` and then serialize it. (I have never used `string_view` so I am not sure if what I am doing is valid)